queries.sql

```diff
@@ -45,3 +45,91 @@ SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;
+
+-- List all animals owned by Dean Winchester that haven't tried to escape.
+SELECT name FROM animals WHERE owner_id = (SELECT id FROM owners WHERE full_name = 'Dean Winchester') AND escape_attempts = 0;
+
+-- Who owns the most animals?
+SELECT full_name, COUNT(*) FROM owners LEFT JOIN animals ON owners.id = animals.owner_id GROUP BY full_name ORDER BY COUNT(*) DESC LIMIT 1;
+
+
+-- QUERIES USING JOIN TABLES
+-- Who was the last animal seen by William Tatcher?
+SELECT animals.name FROM vets 
+  JOIN visits ON vets.id = visits.vets_id
+  JOIN animals ON animals.id = visits.animals_id
+  WHERE vets.name = 'William Tatcher'
+  ORDER BY visits.date_of_visit DESC
+  LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+SELECT COUNT(*) as num_animals_visited from vets
+	JOIN visits ON vets.id = visits.vets_id
+	WHERE vets.name = 'Stephanie Mendez';
+
+-- List all vets and their specialties, including vets with no specialties.
+SELECT vets.name, species.name as specialized_in FROM vets
+	LEFT JOIN specializations ON specializations.vets_id = vets.id
+	LEFT JOIN species ON specializations.species_id = species.id;
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT animals.name as animal_name, visits.date_of_visit FROM animals
+  JOIN visits ON visits.animals_id = animals.id
+  JOIN vets ON vets.id = visits.vets_id
+  WHERE vets.name = 'Stephanie Mendez' AND visits.date_of_visit BETWEEN '2020-04-01' AND '2020-08-30';
+  
+  -- What animal has the most visits to vets?
+SELECT animals.name, COUNT(*) as count FROM animals
+  JOIN visits ON visits.animals_id = animals.id
+  GROUP BY animals.name
+  ORDER BY count DESC
+  LIMIT 1;
+
+-- Who was Maisy Smith's first visit?
+SELECT animals.name FROM visits 
+  JOIN vets ON vets.id = visits.vets_id
+  JOIN animals ON animals.id = visits.animals_id
+  WHERE vets.name = 'Maisy Smith'
+  ORDER BY visits.date_of_visit
+  LIMIT 1;
+  
+  -- Details for most recent visit: animal information, vet information, and date of visit.
+SELECT * FROM visits
+	JOIN animals ON animals.id = visits.animals_id
+	JOIN vets ON vets.id = visits.vets_id
+	ORDER BY date_of_visit
+	LIMIT 1;
+	
+-- How many visits were with a vet that did not specialize in that animal's species?
+SELECT COUNT(*)
+	FROM visits
+	JOIN animals ON animals.id = visits.animals_id
+	JOIN vets ON vets.id = visits.vets_id
+	JOIN specializations ON specializations.vets_id = visits.vets_id
+	WHERE animals.species_id != specializations.species_id;
+
+-- What specialty should Maisy Smith consider getting? Look for the species she gets the most.
+SELECT species.name as species, COUNT(*) FROM visits
+	JOIN vets ON vets.id = visits.vets_id
+	JOIN animals ON animals.id = visits.animals_id
+	JOIN species ON species.id = animals.species_id
+	WHERE vets.name = 'Maisy Smith'
+	GROUP BY species.name
+	ORDER BY count DESC
+	LIMIT 1;
```

Comment:
- [ ] When I run this query, I get an empty row instead of the highest number of species that Maisy Smith attended to the most (Digimon). Kindly return the right result. 👍 

### Your Result

![image](https://user-images.githubusercontent.com/77106962/173126521-6b666ead-3e1c-4ffe-8ea4-04b5c96bb951.png)